src/routers/StackRouter.js

```diff
@@ -138,34 +138,67 @@ export default (
         const childRouter = childRouters[childRoute.routeName];
         if (childRouter) {
           const route = childRouter.getStateForAction(action, childRoute);
+          if (route === null) {
+            return state;
+          }
           if (route && route !== childRoute) {
             return StateUtils.replaceAt(state, childRoute.key, route);
           }
         }
       }
 
-      // Handle push/pop
+      // Handle explicit push navigation action
       if (action.type === NavigationActions.NAVIGATE && childRouters[action.routeName] !== undefined) {
         const childRouter = childRouters[action.routeName];
         let route;
         if (childRouter) {
           const childAction = action.action || NavigationActions.init({ params: action.params });
           route = {
-            ...action,
+            params: action.params,
             ...childRouter.getStateForAction(childAction),
             key: _getUuid(),
             routeName: action.routeName,
           };
         } else {
           route = {
-            ...action,
+            params: action.params,
             key: _getUuid(),
             routeName: action.routeName,
           };
         }
         return StateUtils.push(state, route);
       }
 
+      // Handle navigation to other child routers that are not yet pushed
```

Comment:
Awesome! I thought this was supported with the initial release, but its great to see explicit coverage of a very highly-requested feature